workflows/iuf/operations/management-nodes-rollout/management-m001-rollout.yaml

```
@@ -147,9 +147,20 @@ spec:
                   value: |
                     . /etc/cray/upgrade/csm/myenv
                     if [ ! -f /etc/cray/upgrade/csm/${CSM_REL_NAME}/ncn-m001-bootparameters-update.done ]; then
-                      prepare_images_output=$(echo '{{inputs.parameters.global_params}}' | jq -r '.stage_params["prepare-images"].global["prepare-management-images"]["sat-bootprep-run"].script_stdout')
-                      CFS_CONFIG_NAME=$(echo $prepare_images_output | jq '.images | map(select(.configuration_group_names[] == "Management_Master")) | .[].configuration' | tr -d '"')
-                      IMAGE_ID=$(echo $prepare_images_output | jq '.images | map(select(.configuration_group_names[] == "Management_Master")) | .[].final_image_id' | tr -d '"')
+                      cfs_config="{{=jsonpath(inputs.parameters.global_params,'$.input_params.cfs_configuration_management')}}"
+                      ims_image="{{=jsonpath(inputs.parameters.global_params,'$.input_params.boot_image_management')}}"
+
+                      if [[ -n $cfs_config && -n $ims_image ]]; then
+                        echo "INFO Rebuilding $rebuild_set using passed parameters $ims_image and $cfs_config"
+                        CFS_CONFIG_NAME=$cfs_config
+                        IMAGE_ID=$ims_image
+                      else
+                        echo "INFO Starting rebuild of $rebuild_set using image built in the prepare images stage"
+                        prepare_images_output=$(echo '{{inputs.parameters.global_params}}' | jq -r '.stage_params["prepare-images"].global["prepare-management-images"]["sat-bootprep-run"].script_stdout')
+                        CFS_CONFIG_NAME=$(echo $prepare_images_output | jq '.images | map(select(.configuration_group_names[] == "Management_Master")) | .[].configuration' | tr -d '"')
+                        IMAGE_ID=$(echo $prepare_images_output | jq '.images | map(select(.configuration_group_names[] == "Management_Master")) | .[].final_image_id' | tr -d '"')
```

If you use `jq -r`, you don't need `tr` to remove the double quotes.
```suggestion
                        CFS_CONFIG_NAME=$(echo $prepare_images_output | jq -r '.images | map(select(.configuration_group_names[] == "Management_Master")) | .[].configuration')
                        IMAGE_ID=$(echo $prepare_images_output | jq -r '.images | map(select(.configuration_group_names[] == "Management_Master")) | .[].final_image_id')
```